src/service/internal_tables_access.h

```diff
@@ -331,10 +333,29 @@ namespace ccf
       if (service->has())
       {
         const auto prev_service_info = service->get();
+
         auto previous_service_identity = tx.wo<ccf::PreviousServiceIdentity>(
           ccf::Tables::PREVIOUS_SERVICE_IDENTITY);
         previous_service_identity->put(prev_service_info->cert);
 
+        auto previous_identity_endorsement =
+          tx.wo<ccf::PreviousServiceIdentityEndorsement>(
+            ccf::Tables::PREVIOUS_SERVICE_IDENTITY_ENDORSEMENT);
+        try
+        {
+          const auto prev_ident_endorsement = cose_sign1(
+            service_key,
+            {}, // TO DO headers,
+            prev_service_info->cert.raw());
+          previous_identity_endorsement->put(prev_ident_endorsement);
+        }
+        catch (const ccf::crypto::COSESignError& e)
+        {
+          LOG_FAIL_FMT("Failed o sign previous service identity: {}", e.what());
+          throw; // TO DO catch re-throw in frontent? Or change to log and still
```

Comment:
I'm inclined to make it fail via `throw` to propagate the error back and let the caller decide.